onnxruntime/core/providers/cpu/nn/layer_norm_impl.cc

```diff
@@ -237,9 +234,11 @@ Status LayerNormImpl::PrePack(const Tensor& tensor, int input_idx, AllocatorPtr
 
   is_packed = false;
   if (input_idx == 1) {  // scale
-    ConvertMLFloat16ToFloatIfNeeded(tensor, alloc, scale_fp32_, is_packed);
+    prepacked_scale_fp32_size_ = tensor.Shape().Size();
```

Comment:
explicit cast from int64 to size_t.

Reply:
do you suggest to just cast or using cast with check (eg. SafeInt or gsl::narrow)?

Reply:
SafeInt is safer than static_cast.
I think static_cast is fine for this case (since it is less likely user could pass in a tensor that is larger than max uint32 in x86)